safewalk2-platform/README.md

```diff
@@ -0,0 +1,322 @@
+6. Assuming that the credentials have been verified successfully (i.e. Access-Accept) the user is granted access
+to the application;
+
+#####* Registration-Gateway
+For facilitating the over-the-air registration method of mobile applications.
+
+#####* Safewalk server authentication api
+A proxy to the Safewalk server authentication api for external applications that do not support standard authentication protocols.
+
+**General authentication flow:**
+1. User generates a code (be it its static password, One-Time-Password from a mobile app/email/sms or from
+a physical device);
+2. User browses to the address of the organization NAS (that is equipped with a RADIUS client), where he is
+prompted to enter its credentials;
+3. The credentials supplied by the user are checked on the Safewalk server;
+4. The Safewalk server validates the code and return a reply to the NAS (one of Access-Accept/AccessReject/Access-Challenge);
+5. Assuming that the credentials have been verified successfully (i.e. Access-Accept) the user is granted access
+to the relevant application;
+
+
+
+## Prerequisites
+
+To get use this Safewalk2 platform you'll need to copy or upload the VM's VHD images to your storage account.
+
+
+### Uploading the Safewalk Server and Safewalk Gateway VHD images
+
+The first thing you will need to do before you can deploy Safewalk using the ARM template is to copy or upload the Safewalk Server and Safewalk Gateway VHD images to your Azure subscription by following the steps below:
+
+Login into the Azure portal (https://portal.azure.com)
+
+Create or select a blob storage account on the **same region** where you plan to deploy the Safewalk2 framework. It's recommended that the VHD Images live in a separate resource group in case you need to delete/move the system components and keep the images for future deployments.
+Then add a container to copy/upload the VHD images into.
+
+You can install in your machine the storage tool **AzCopy** to make a copy of the custom VHD images in your container. <a href="http://aka.ms/downloadazcopy" target="_blank">Download and install the latest version of AzCopy</a>
+
+After installing **AzCopy** on Windows using the installer, open a command window and navigate to the AzCopy installation directory on your computer - where the AzCopy.exe executable is located (normally at **%ProgramFiles(x86)%\Microsoft SDKs\Azure\AzCopy**)
+
+Use the following command replacing the **{dest_url}** with your container information (copy it from the container properties), and provide the storage account access key2 **{dest_key2}** (copy it from your storage account **Access Keys** section).
+
+```PowerShell
+AzCopy /Source:https://safewalk2.blob.core.windows.net/images /Dest:{dest_url} /SourceKey:i4dCa1J6O1TriXgGFrS2V5N/Zjw6GU9JR8dckydWHWaodLWDmHoDFQA0lrEuDLfKZgE0owpwTPThXMrmYLIGtQ== /DestKey:{dest_key2} /S
```

Comment:
Thanks for your feedback. As we are using custom VHD images, How is the best way to provide it to the customer? There is an automatic method to copy the images or can be used from a remote location (not in the same subscription)?

Reply:
One option is to put the VHDs in the Azure Marketplace (https://azuremarketplace.microsoft.com/en-us/sell) the other is to use VM extensions to configure the VM (instead of the VHD)

Reply:
We are planning to go to the market next year (we still need to review the online business model based on resource consumption). In the meanwhile can I keep the instructions to use our vhd images (by downloading from our read only storage and then upload it to the customer subscription)? I know it's a manual mechanism but I find it hard to write a complete system setup from scratch using scripts. Thanks

Reply:
Unfortunately, no - we need to be able to have some validation of the workload that this is going to run...  So if we can't validate the image via marketplace, we can do so via the code that configures the workload.